### DISCUSSION TITLE
submit defaultvalue when required is set

### DISCUSSION THREAD

**Discussion by erganmedia** [react-hook-form/react-hook-form#2287] [Q&A] [answered]
Hey Guys,

My question is: why am I not able to submit the form if a value is required and default value is set?
I try to implement a edit functionality where my values are set as default values and if the user decides to just change one value the other input fields prevent me from submitting the form since they are not touched/dirty.

I prepared an simple expamle for you guys.. just don't change anything and click on submit.
[codesandbox](https://codesandbox.io/s/react-hook-form-defaultvalues-v6-cjm51?file=/src/index.js)

Is there a way of doing this?.. what am I doing wrong? :/
thank you.

**Comment by bluebill1049** [comment, marked as answer]
you are after `isDirty` formState right?

**Reply by erganmedia** [reply]
I mean.. I don't really care if the user has changed anything.. I just want to allow him to submit the form with all the default values set by me.

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/react-hook-form-defaultvalues-v6-11huh?file=/src/index.js

take a look the `register` at the doc, you are doing a custom register

**Reply by erganmedia** [reply]
that worked for me.. thank you :)
